returnn/tf/layers/base.py

```diff
@@ -1478,6 +1478,11 @@ def __init__(self, layer_class, layer_desc, **kwargs):
     self.layer_class_ = layer_class
     self.layer_desc = layer_desc
 
+    register_as_extern_data = layer_desc.get("register_as_extern_data")
+    if register_as_extern_data:
+      self.network.extern_data.extra_added_keys.add(register_as_extern_data)
+      self.network.extern_data.data[register_as_extern_data] = self.output
```

Comment:
I don't understand. Why is this needed here?

Reply:
> Usually this happens in `LayerBase.post_init()`, however this is not called for DataNotAvailableLayer.

Why? And shouldn't we fix that? This again sounds like you do a workaround here.

Reply:
@patrick-wilken Is there any unwanted side effect in calling `post_init()` on the `DataNotAvailableLayer`? I am not sure what happens if we call `self.batch_norm` on an output which has no defined placeholder. if there are no side effects, we could call `post_init` before returning the `DataNotAvailableLayer`, if there are, this seems to be the correct solution.

Reply:
If `LayerBase.post_init()` does not work on `DataNotAvailableLayer`, then why do another workaround at a different place? Why not just define `DataNotAvailableLayer.post_init()`? Or fix `LayerBase.post_init()`?